src/Microsoft.ComponentDetection.Detectors/go/GoComponentDetector.cs

```diff
@@ -104,22 +105,31 @@ private async Task<bool> UseGoCliToScan(string location, ISingleFileComponentRec
             var projectRootDirectory = Directory.GetParent(location);
             record.ProjectRoot = projectRootDirectory.FullName;
 
-            var isGoAvailable = await CommandLineInvocationService.CanCommandBeLocated("go", null, workingDirectory: projectRootDirectory, new List<string> { "version" }.ToArray());
+            var isGoAvailable = await CommandLineInvocationService.CanCommandBeLocated("go", null, workingDirectory: projectRootDirectory, new[] { "version" });
             record.IsGoAvailable = isGoAvailable;
 
             if (!isGoAvailable)
             {
                 return false;
             }
 
+            var goDependenciesProcess = await CommandLineInvocationService.ExecuteCommand("go", null, workingDirectory: projectRootDirectory, new[] { "list", "-m", "-json", "all" });
```

Comment:
Do you have an expected execution time for this? I know `go mod graph` has caused issues for us in the past.

Reply:
The problem is not `go mod graph` itself IIRC, the problem is when users have `go.sum` and `go.mod` files in their repo, but they actually aren't building it. Go CLI will try to fetch the packages behind the scenes if they aren't in the system already, which will add time to any automated process. Execution time should be the about the same, but now `go list` takes the hit of fetching the packages, rather than `go mod graph`. 

At the end of the day, we are using this command because we do not have any other native way to accurately detect components in Go. I tested with a big repo reporting ~730 dependencies using only `go mod graph` vs the latest changes that only pick up 137 dependencies that are actually included in their Go binaries.